src/serverless.js

```diff
@@ -145,6 +145,192 @@ class ServerlessComponent extends Component {
+        alias_unit[aliases[i].Name].weight = JSON.stringify(weight)
+
+        listAlias.alias.push(alias_unit)
+      }
+      console.log(listAlias)
+      return listAlias
+    } catch (e) {
+      return {
+        requestId: e.reqId,
+        message: e.message
+      }
+    }
+  }
+
+  async create_alias(inputs) {
+    try {
+      const credentials = this.getCredentials()
+      const region = inputs.region || CONFIGS.region
+
+      const alias_params = prepareAliasInputs(inputs)
+
+      if (alias_params.isPramasError) {
+        return {
+          message: alias_params.message
+        }
+      }
+
+      console.log(`Creating alias for function ${inputs.function}...`)
+      const scf = new Scf(credentials, region)
+      await scf.createAlias(alias_params)
+      console.log(`Creating alias for function ${inputs.function}...`)
+
+      const aliasOutput = await scf.getAlias(alias_params)
+
+      return aliasOutput
+    } catch (e) {
+      return {
+        requestId: e.reqId,
+        message: e.message
+      }
+    }
+  }
+
+  async update_alias(inputs) {
+    try {
+      const credentials = this.getCredentials()
+      const region = inputs.region || CONFIGS.region
+
+      const alias_params = prepareAliasInputs(inputs)
+
+      if (alias_params.isPramasError) {
+        return {
+          ErrMsg: alias_params.message
+        }
+      }
+
+      console.log(`Updating alias for function ${inputs.function}...`)
+      const scf = new Scf(credentials, region)
+
+      await scf.updateAlias(alias_params)
+      console.log(`Updated alias for function ${inputs.function}...`)
+
+      const aliasOutput = await scf.getAlias(alias_params)
+
+      return aliasOutput
+    } catch (e) {
+      return {
+        requestId: e.reqId,
+        message: e.message
+      }
+    }
+  }
+
+  async delete_alias(inputs) {
+    try {
+      const credentials = this.getCredentials()
+      const region = inputs.region || CONFIGS.region
+
+      const alias_params = {}
+      alias_params.functionName = inputs.function
+      alias_params.aliasName = inputs.name
+      alias_params.namesapce = inputs.namespace
+
+      console.log(`delete alias for function ${inputs.function}...`)
+      const scf = new Scf(credentials, region)
+
+      const scfOutput = await scf.deleteAlias(alias_params)
+      console.log(`deleted alias for function ${inputs.function}...`)
+      return scfOutput
+    } catch (e) {
+      return {
+        requestId: e.reqId,
+        message: e.message
+      }
+    }
+  }
+
+  async publish_ver(inputs) {
+    try {
+      const credentials = this.getCredentials()
+      const region = inputs.region || CONFIGS.region
+
+      const publish_params = {}
+      publish_params.functionName = inputs.function
+      publish_params.namespace = inputs.namespace
+      publish_params.description = inputs.description
+
+      const scf = new Scf(credentials, region)
+
+      const scfOutput = await scf.publishVersion(publish_params)
```

Comment:
Publish version of SCF is an asynchronous flow, maybe you should wait for the `Active` status of SCF.